Draft-Accepted/RFCNNNN-Install-PowerShell.md

```diff
@@ -0,0 +1,175 @@
+
+### Solution
+
+To solve these issues, the proposal is to ship a new command in Windows that would:
+
+- Be locally discoverable
+- Perform a simple download and install of the latest LTS release of PowerShell
+- Include parameter options to select Stable, Preview, or LTS releases
+
+## Goals/Non-Goals
+
+Goals:
+
+- By default, download and install the latest LTS release to the users CurrentUser Scope
+  - CurrentUser - default installation scope location $env:LOCALAPPDATA\Microsoft\pwsh. If the path
+    exists, remove and replace.
+  - AllUsers - Administrative users may explicit select this scope. AllUsers requires administrative
+    elevation and performs a silent install using the MSI defaults.
+- Allow users to use remoting for install at scale with Invoke-Command
+- Allow advanced users to specify channel Stable/Preview/LTS
+
+Non-goals:
+
+- Removal of PowerShell (Remove-PowerShell7)
+- Updating of PowerShell (Update-PowerShell7)
+- User selected versions prior to current (No version parameter)
+- Disconnected scenarios - Users that are air-gapped may download the appropriate release from GitHub.
+- Handling of additional MSI switch options - i.e. Enable PSRemoting
+- Cross platform support
+
+## Specification
+
+Command name: install-powershell7
+
+The default behavior is to install to CurrentUser scope, with the latest LTS
+release of PowerShell. If the user has administrative
+privilege, they can choose to install to the scope of AllUsers.
+
+- The package install location - $env:LOCALAPPDATA\Microsoft\pwsh
+  - If this path already exists, remove and replace.
+- The installer should update the Users PATH for pwsh
+- The installer should enable Microsoft Updates for scope AllUsers.
+
+The following parameters can be added by experienced PowerShell users to customize the installation.
+
+- Parameter: Scope - overrides the default of CurrentUser and allows the user to specify the
+  installation location AllUsers.
+- Parameter: Channel - options Stable/Preview/LTS - allows the user to select if they want the
+  current preview or LTS release.
+
+## Syntax
+
+The command `install-powershell7` is lower case and supports `-/--` switch options to be consistent
+with other command-line tools.
+
+```syntax
+install-powershell7  [-channel <option>] [-scope <option>]
+```
+
+The `--channel <option>` must be one of the following:
+
+- `--channel preview`
+- `--channel stable`
+- `--channel lts`
+
+If you don't provide a channel option, the command defaults to `-channel lts`.
+
+The `--scope <option>` must be one of the following:
+
+- `--scope allusers`
+- `--scope currentuser`
+
+If you don't provide a scope option, the command defaults to `--scope currentuser`.
+
+Scope selection has an impact on receiving auto-updates from Microsoft Update:
+
+- `--scope allusers` is an MSI-based installation with Microsoft Update enabled.
+- `--scope currentuser` is a ZIP-based installation. Users will need to manually
```

Comment:
Do we need to have a different `$env:POWERSHELL_DISTRIBUTION_CHANNEL` value for ZIP-based install?

Reply:
We do not set a `POWERSHELL_DISTRIBUTION_CHANNEL` for ZIP based installs now, but we can add through our tool. There could be complications if there are multiple versions installed. For example, the enterprise has LTS deployed via MSI, but the employee installs preview via this tool. Do we overwrite the value?

Reply:
Also, for MSI installs we might want to have a different value for distribution channel to signify MSI installed via InstallPowerShell7.exe

Reply:
This tool should not do anything with the PowerShell distribution channel.